tests/rptest/tests/cluster_metrics_test.py

```diff
@@ -93,11 +83,34 @@ def _get_cluster_metrics(
             self, node: ClusterNode) -> Optional[dict[str, MetricSamples]]:
         """
         Get all the cluster level metrics exposed by a specified
-        node in the cluster.
-        """
+        node in the cluster. If 'node' is the current controller,
+        wait briefly for the metrics to become available.
+        """
+        def get_metrics_from_node_sync(pattern: str):
```

Comment:
we should wait the same amount of time before making a decision whether the node is controller or not. imagine a follower returns metrics but it has the same lag as a controller, if we query it once we won't notice it. Can you please change it to something like this?

```
def get_metrics_from_node_sync(pattern: str):
    samples = self.redpanda.metrics_sample(
        pattern, [node], MetricsEndpoint.PUBLIC_METRICS)
    success = samples is not None
    return success, samples

def get_metrics_from_node(pattern: str):
    if expect_metrics:
        return wait_until_result(
            lambda: get_metrics_from_node_sync(pattern),
            timeout_sec=2,
            backoff_sec=.1)
    else:
        try:
            wait_until_result(
                lambda: get_metrics_from_node_sync(pattern),
                timeout_sec=2,
                backoff_sec=.1)
        except TimeoutError:
            return None
```

Reply:
Not waiting for the followers is an optimisation to keep the test short. We query the controller leader before the followers, which means that by the time we get to a follower we've already waited on the metrics from the leader. I'm happy to go with what you're proposing though. The test will take longer, but it's easier to see through it.

Reply:
I ended up removing this optimisation and doing what you suggested. In order to keep the tests short I've added a new method to the Redpanda service that fetches multiple metrics at once. This means that now, we wait on the full set of metrics instead of individual ones.